AP_Scripting: add bindings for E-Stop, Motor interlock and safety state

IamPete1 opened the pull request:
Adds bindings to allow scripts to be more aware of the safety critical states of the vehicle. If motors are in interlock or E-Stop is active then the script should stop outputting to any motors it might be controlling. If the safety state is disarmed it should not try and output stuff that is not allowed through the safety mask. 

Usefull for some of the new EFI/generator/governor scripts. If safety is active then the engine should not be started. If E-Stop is active then the engine should be stopped.

I ended up adding a passthrough method for safety state to servo channels `hal.util->safety_switch_state()`. We could add a `hal.util` binding like we have for `hal.gpio` but I don't think there would ever be any other methods. Servo seems a logical place for it to live in scripting.

rmackay9 reviewed (approved):
LGTM.  It's been tested of course?  I'm sure it has but..

IamPete1 commented:
Rebased and made a example out of my test script.

IamPete1 commented:
Rebased and moved the safety state binding to a manual, this allows us to get the safety state from the `SRV_Channels` table without needing a helper.
